packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -104,24 +168,330 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
     errorMessage: styles?.systemMessage
   });
 
-  const onTexFieldKeyDown = useCallback(
+  /* @conditional-compile-remove(mention) */
+  const onSuggestionSelected = useCallback(
+    (suggestion: Mention) => {
+      let selectionEnd = textFieldRef?.current?.selectionEnd || -1;
+      if (selectionEnd < 0) {
+        selectionEnd = 0;
+      } else if (selectionEnd > inputTextValue.length) {
+        selectionEnd = inputTextValue.length;
+      }
+      const oldPlainText = inputTextValue;
+      const mention = htmlStringForMentionSuggestion(suggestion, localeStrings);
+
+      // update plain text with the mention html text
+      const newPlainText =
+        inputTextValue.substring(0, currentTriggerStartIndex) + mention + inputTextValue.substring(selectionEnd);
+      const triggerText = mentionLookupOptions?.trigger ?? defaultMentionTrigger;
+      // update html text with updated plain text
+      const [updatedHTML] = updateHTML(
+        textValue,
+        oldPlainText,
+        newPlainText,
+        tagsValue,
+        currentTriggerStartIndex,
+        selectionEnd,
+        mention,
+        triggerText
+      );
+      const displayName = getDisplayNameForMentionSuggestion(suggestion, localeStrings);
+      // Move the caret in the text field to the end of the mention plain text
+      setCaretIndex(selectionEnd + displayName.length);
+      setCurrentTriggerStartIndex(-1);
+      updateMentionSuggestions([]);
+      setActiveSuggestionIndex(undefined);
+      onChange && onChange(undefined, updatedHTML);
+    },
+    [
+      textFieldRef,
+      inputTextValue,
+      currentTriggerStartIndex,
+      mentionLookupOptions?.trigger,
+      onChange,
+      textValue,
+      tagsValue,
+      /* @conditional-compile-remove(mention) */
+      updateMentionSuggestions,
+      /* @conditional-compile-remove(mention) */
+      localeStrings
+    ]
+  );
+
+  const onTextFieldKeyDown = useCallback(
     (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
       // Uses KeyCode 229 and which code 229 to determine if the press of the enter key is from a composition session or not (Safari only)
       if (ev.nativeEvent.isComposing || ev.nativeEvent.keyCode === 229 || ev.nativeEvent.which === 229) {
```

Comment:
Could be constant put into constant file

Reply:
Will be part of the upcoming refactor task we chatted about.